src/query/dblink_scan.c

```diff
@@ -301,6 +333,147 @@ dblink_make_date_time_tz (T_CCI_U_TYPE utype, DB_VALUE * value_p, T_CCI_DATE_TZ
   return error;
 }
 
+static int
+dblink_bind_param (DBLINK_SCAN_INFO * scan_info, VAL_DESCR * vd)
+{
+  int i, ret;
+  T_CCI_PARAM_INFO *param;
+  T_CCI_A_TYPE a_type;
+  T_CCI_U_TYPE u_type;
+  void *value;
+  double adouble;
+  int month, day, year;
+  int hh, mm, ss, ms;
+  DB_TIMESTAMP *timestamp;
+  DB_DATETIME *datetime;
+  DB_DATETIME dt_local;
+  TZ_ID *zone_id;
+  DB_DATE date;
+  DB_TIME time;
+  T_CCI_DATE cci_date;
+
+  unsigned char type;
+
+  for (i = 0; i < vd->dbval_cnt; i++)
+    {
+      value = &vd->dbval_ptr[i].data;
+      type = vd->dbval_ptr[i].domain.general_info.type;
+      switch (vd->dbval_ptr[i].domain.general_info.type)
+	{
+	case DB_TYPE_INTEGER:
+	  a_type = CCI_A_TYPE_INT;
+	  u_type = CCI_U_TYPE_BIGINT;
+	  break;
+	case DB_TYPE_BIGINT:
+	  a_type = CCI_A_TYPE_BIGINT;
+	  u_type = CCI_U_TYPE_BIGINT;
+	  break;
+	case DB_TYPE_NUMERIC:
```

Comment:
If you modify it as follows, the type variable is not needed.

```
	case DB_TYPE_NUMERIC:
           {
	      numeric_coerce_num_to_double (db_locate_numeric (&vd->dbval_ptr[i]),
					    vd->dbval_ptr[i].domain.numeric_info.scale, &adouble);
	      value = &adouble;
	    }
	case DB_TYPE_DOUBLE:
	case DB_TYPE_FLOAT:
	  a_type = CCI_A_TYPE_DOUBLE;
	  u_type = CCI_U_TYPE_DOUBLE;	  
	  break;
```

Reply:
ok, I agree.